src/http/web_socket.cr

```diff
@@ -100,52 +101,86 @@ class HTTP::WebSocket
     @ws.close(message)
   end
 
-  def run
+  record Ping, message : String
+  record Pong, message : String
+  record Text, message : String
+  record Binary, message : Bytes
+  record Close, message : String
```

Comment:
Does this need to be five different types with almost the same layout? Maybe it would be better to have only a single struct with an enum property to designate the type of message?

Reply:
Interesting. Do you think it makes sense to reuse the Protocol::Opcodes? And for the message property, it would have a type of String | Bytes?

Reply:
Yeah, message type is a bit tricky. As long as `String` support ordinary non-utf8 data it would suffice alone, but we can't count on that...

Reply:
I would store it internally as a union and expose it through two properties that cast the value with `as`.